### PR TITLE
Replace glog with zerolog.

### PR DESCRIPTION
**Pull request by jack0**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -231,7 +231,7 @@ type ChatPlugin struct{ *network.Plugin }
 func (state *ChatPlugin) Receive(ctx *network.PluginContext) error {
     switch msg := ctx.Message().(type) {
         case *messages.ChatMessage:
-            glog.Infof("<%s> %s", ctx.Client().ID.Address, msg.Message)
+            log.Info().Msgf("<%s> %s", ctx.Client().ID.Address, msg.Message)
```
Comment (doug-perlin) [inline README.md]: Is there going to be another PR to convert these logs into key value pairs? Or is this just a first draft

File: examples/chat/main.go [hunk 2]
```diff
@@ -57,7 +54,7 @@ func main() {
 
 	net, err := builder.Build()
 	if err != nil {
-		glog.Fatal(err)
+		log.Fatal().Err(err)
```
Comment (doug-perlin) [inline examples/chat/main.go]: for warning/error/fatal, we should add file and line number to log to help debug:
```
log.Logger = log.With().Caller().Logger()
log.Info().Msg("hello world")

// Output: {"level": "info", "message": "hello world", "caller": "/go/src/your_project/some_file:21"}
```
https://github.com/rs/zerolog#add-file-and-line-number-to-log

File: internal/test/protobuf/test.proto [hunk 3]
```diff
@@ -19,4 +19,5 @@ option (gogoproto.testgen_all) = true;
 
 message TestMessage {
     string message = 1;
+    int32 duration = 2;
```
Comment (cristaloleg) [inline internal/test/protobuf/test.proto]: Can we limit scope of this PR to only zerolog? And left any logic changes to another PR ?

Comment (cristaloleg) [reply internal/test/protobuf/test.proto]: Ah, it is caused by #158. nevermind

File: examples/chat/main.go (hunk 2 above)
Comment (jack0) [reply examples/chat/main.go]: it doesn't seem to work that well:
```
{"level":"info","time":"2018-09-19T10:29:59-04:00","caller":"/Users/jack/go/pkg/mod/github.com/rs/zerolog@v1.8.0/event.go:116","message":"Setting up NAT traversal for address: tcp://127.0.0.1:49197"}
{"level":"warn","error":"no NAT found","time":"2018-09-19T10:30:09-04:00","caller":"/Users/jack/go/repos/noise/network/nat/plugin.go:43","message":"Unable to discover gateway"}
{"level":"info","time":"2018-09-19T10:30:09-04:00","caller":"/Users/jack/go/pkg/mod/github.com/rs/zerolog@v1.8.0/event.go:116","message":"Listening for peers on tcp://127.0.0.1:49197."}
{"level":"info","time":"2018-09-19T10:30:09-04:00","caller":"/Users/jack/go/pkg/mod/github.com/rs/zerolog@v1.8.0/event.go:116","message":"Setting up NAT traversal for address: tcp://127.0.0.1:49375"}
{"level":"warn","error":"no NAT found","time":"2018-09-19T10:30:19-04:00","caller":"/Users/jack/go/repos/noise/network/nat/plugin.go:43","message":"Unable to discover gateway"}
{"level":"info","time":"2018-09-19T10:30:19-04:00","caller":"/Users/jack/go/pkg/mod/github.com/rs/zerolog@v1.8.0/event.go:116","message":"Listening for peers on tcp://127.0.0.1:49375."}
```

Comment (jack0) [reply examples/chat/main.go]: related to this issue that was fixed july https://github.com/rs/zerolog/pull/89

v1.8 that we're using is from may, but they just released v1.9.0 7 hours ago.. updating the version fixes the line number issue

File: README.md (hunk 1 above)
Comment (jack0) [reply README.md]: done

**Comment by jack0** [comment]
when i add line breaks to the log messages without existing code coverage each line decreases it further ;)

File: network/network.go [hunk 4]
```diff
@@ -38,6 +41,14 @@ var (
 	_ NetworkInterface = (*Network)(nil)
 )
 
+func init() {
+	log.Logger = log.With().Caller().Logger()
+	// prettify if terminal is a console
+	if terminal.IsTerminal(int(os.Stdout.Fd())) {
```
Comment (doug-perlin) [inline network/network.go]: Will this always be a terminal? When would it be false?

Comment (doug-perlin) [inline network/network.go]: Will this always be a terminal? When would it be false?

Comment (jack0) [reply network/network.go]: its false if its running as tests
